spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -2739,8 +2740,13 @@ def fix_indent_smart(self, forward=True, comment_or_string=False):
 
         # TODO untangle this block
         if prevline and not bracket_stack and not prevtext.endswith(':'):
-            cur_indent = self.get_block_indentation(block_nb - 1)
-            is_blank = not self.get_text_line(block_nb - 1).strip()
+            above_block_nb = block_nb - 1
+            # If the block above is empty, check the previous one
+            if (ignore_empty_line and self.get_text_line(above_block_nb) == ''

```

Comment:
This could be a while to allow for any number of empty lines. I think a single empty line is better.

Reply:
I think this is fine, i.e. pressing backspace dedents 4 spaces if the previous line is empty (as requested on issue #10146) but it goes two the first column after two or more empty lines.

@bcolsen, what do you think?

Reply:
Now that removing trailing spaces only works when the option is turned on in our Preferences, I think we can merge this one as it is.

Reply:
(That was done in PR #10261).